### DISCUSSION TITLE
GitHub Action not starting

### DISCUSSION THREAD

**Discussion by SeanKilleen** [community/community#25143] [Actions]
<p>Hi all! Hoping you can help me sort this out.</p>
<p><strong>Background</strong></p>
<p>I am attempting to create a GitHub Actions workflow and I am unsure why it isn’t being triggered or shown; I’m not getting any feedback as to why. I’m sure I must be missing something.</p>
<p>YAML file is on the master branch: <a href="https://github.com/CompositionalIT/farmer/blob/master/.github/workflows/gh-pages.yml" rel="nofollow noopener">https://github.com/CompositionalIT/farmer/blob/master/.github/workflows/gh-pages.yml</a></p>
<h2>Expected Behavior</h2>
<p>* Push a commit into the <code>docs</code> branch (e.g. a PR being merged or a commit being made)</p>
<p>* Action is triggered</p>
<p>* Docs branch is pulled, built using Hugo, and pushed into the <code>gh-pages</code> branch</p>
<p>* GitHub pages then serves the content.</p>
<h2>Actual Behavior</h2>
<p>* Make a change to the <code>docs</code> branch (have tried PR merges and direct commits)</p>
<p>* Nothing happens</p>
<p>I can’t figure out what is off about the Actions workflow that is causing it to be unable to be triggered. </p>
<p>Can anyone provide some insight? <a href="https://github.com/CompositionalIT/farmer/blob/master/.github/workflows/gh-pages.yml" rel="nofollow noopener">https://github.com/CompositionalIT/farmer/blob/master/.github/workflows/gh-pages.yml</a></p>

**Comment by managedkaos** [comment, marked as answer]
<p>Hi <a class="mention" href="/seankilleen">@seankilleen</a> !  I saw your request and wanted to help out.</p>
<p>I looks like your workflow is not getting triggered because the <code>docs</code> branch didn’t have a workflow.  In fact, the worflow file is only  on the <code>master</code> branch.</p>
<p>I submitted a PR: <a href="https://github.com/CompositionalIT/farmer/pull/94" rel="nofollow noopener">https://github.com/CompositionalIT/farmer/pull/94</a></p>
<p>Hope that it helps!</p>
<p>-M</p>

**Comment by SeanKilleen** [comment]
<p>Thank you <a class="mention" href="/managedkaos">@managedkaos</a>!</p>
<p>Can you help me understand why an action on the master branch that is scoped to the docs branch not have this affect? Is it that a workflow by the same name needs to exist in all relevant branches?</p>

**Comment by managedkaos** [comment]
<p><a href="https://help.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow#creating-a-workflow-file" title="Creating a workflow file" rel="nofollow noopener">Creating a workflow file</a><br>
Commit your changes in the workflow file to the branch where you want your workflow to run.</p>

**Comment by SeanKilleen** [comment]
<p>Very helpful. Thank you so much again!</p>
